cmd/agent/app/jmx.go

```diff
@@ -0,0 +1,184 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package app
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/jmxfetch"
+	"github.com/spf13/cobra"
+)
+
+var (
+	jmxCmd = &cobra.Command{
+		Use:   "jmx",
+		Short: "",
+		Long:  ``,
+	}
+
+	jmxListCmd = &cobra.Command{
+		Use:   "list",
+		Short: "List attributes matched by JMXFetch.",
+		Long:  ``,
+	}
+
+	jmxCollectCmd = &cobra.Command{
+		Use:   "collect",
+		Short: "Start the collection of metrics based on your current configuration and display them in the console.",
+		Long:  ``,
+		Run:   doJmxCollect,
+	}
+
+	jmxListEverythingCmd = &cobra.Command{
+		Use:   "everything",
+		Short: "List every attributes available that has a type supported by JMXFetch.",
+		Long:  ``,
+		Run:   doJmxListEverything,
+	}
+
+	jmxListMatchingCmd = &cobra.Command{
+		Use:   "matching",
+		Short: "List attributes that match at least one of your instances configuration.",
+		Long:  ``,
+		Run:   doJmxListMatching,
+	}
+
+	jmxListLimitedCmd = &cobra.Command{
+		Use:   "limited",
+		Short: "List attributes that do match one of your instances configuration but that are not being collected because it would exceed the number of metrics that can be collected.",
+		Long:  ``,
+		Run:   doJmxListLimited,
+	}
+
+	jmxListCollectedCmd = &cobra.Command{
+		Use:   "collected",
+		Short: "List attributes that will actually be collected by your current instances configuration.",
+		Long:  ``,
+		Run:   doJmxListCollected,
+	}
+
+	jmxListNotMatchingCmd = &cobra.Command{
+		Use:   "not-matching",
+		Short: "List attributes that don’t match any of your instances configuration.",
+		Long:  ``,
+		Run:   doJmxListNotCollected,
+	}
+
+	checks = []string{}
+)
+
+func init() {
+	// attach list and collect commands to jmx command
+	jmxCmd.AddCommand(jmxListCmd)
+	jmxCmd.AddCommand(jmxCollectCmd)
+
+	//attach list commands to list root
+	jmxListCmd.AddCommand(jmxListEverythingCmd, jmxListMatchingCmd, jmxListLimitedCmd, jmxListCollectedCmd, jmxListNotMatchingCmd)
+
+	jmxListCmd.PersistentFlags().StringSliceVar(&checks, "checks", []string{"jmx"}, "JMX checks (ex: jmx,tomcat)")
+	jmxCollectCmd.PersistentFlags().StringSliceVar(&checks, "checks", []string{"jmx"}, "JMX checks (ex: jmx,tomcat)")
```

Comment:
It's a good idea to allow specifying checks, but the default should be that it collects from all configured jmx checks (unless it's very complex to implement)